src/standalone/Vegetation/canopy_parameterizations.jl

```diff
@@ -1099,3 +1108,459 @@ end
+        0.161913    0.257399     0.0        0.0        0.0         0.0;
+       -0.0325372   0.0          0.0        0.0698452  0.0         0.0;
+        0.0         0.0          0.0        0.0        0.00872102  0.0;
+        0.0         0.0          0.0       -0.00435673 0.0        -0.000593264
+    ])
+
+    # Compute μ1 (Eq. 12 & Table 3)
+    μ1 = FT(0.0)
+    ctbar = (FT(1.0) / tbar) - FT(1.0)
+    for i in 1:6
+        coef1 = ctbar^(i - 1)
+        coef2 = FT(0.0)
+        for j in 1:7
+            coef2 += h_array[j, i] * (ρbar - FT(1.0))^(j - 1)
+        end
+        μ1 += coef1 * coef2
+    end
+    μ1 = exp(ρbar * μ1)
+
+    μ_bar = μ0 * μ1       # Eq. 2
+    μ = μ_bar * μ_ast     # Eq. 1
+    return FT(μ)          # Pa s
+end
+
+
+"""
+    compute_viscosity_ratio(
+        T::FT, 
+        p::FT
+    ) where {FT}
+
+    Computes η*, the ratio of the viscosity of water at temperature T and pressure p
+    to the viscosity of water at STP. 
+"""
+function compute_viscosity_ratio(T::FT, p::FT) where {FT} 
+    η25 = viscosity_h2o(FT(298.15), FT(101325.0))
+    ηstar = viscosity_h2o(T, p) / η25
+    return FT(ηstar)
+end
+
+
+"""
+    po2(P_air::FT) 
+
+    Computes the partial pressure of O2 in the air (Pa) given atmospheric pressure. Assumes a 
+    constant mixing ratio of 21000 / 101325. 
+"""
+function po2(P_air::FT) where {FT} 
+    return FT(0.2095) * P_air 
+end
+
+"""
+    co2_compensation_p(
+        T::FT,
+        To::FT,
+        p::FT,
+        R::FT, 
+        ΔHΓstar::FT = FT(37830),
+        Γstar25::FT = FT(4.332) 
+    ) where {FT}
+
+    Computes the CO2 compensation point (`Γstar`), in units Pa, as a function of temperature T (K)
+    and pressure p (Pa). See Equation B5 of Stocker et al. (2020). 
+"""
+function co2_compensation_p(
+    T::FT,
+    To::FT,
+    p::FT,
+    R::FT, 
+    ΔHΓstar::FT = FT(37830),
+    Γstar25::FT = FT(4.332) 
+) where {FT}
+    Γstar = Γstar25 * p / FT(101325.0) * arrhenius_function(T, To, R, ΔHΓstar)
+    return Γstar
+end
+
+"""
+    compute_Kmm(
+        T::FT, 
+        p::FT, 
+        Kc25::FT, 
+        Ko25::FT, 
+        ΔHkc::FT, 
+        ΔHko::FT, 
+        To::FT, 
+        R::FT
+    ) where {FT}
+
+    Computes the effective Michaelis-Menten coefficient for Rubisco-limited photosynthesis (`Kmm`),
+    in units Pa, as a function of temperature T (K), atmospheric pressure p (Pa), and constants:
+    Kc25 (Michaelis-Menten coefficient for CO2 at 25 °C), Ko25 (Michaelis-Menten coefficient for O2 at 25 °C),
+    ΔHkc (effective enthalpy of activation for Kc), ΔHko (effective enthalpy of activation for Ko),
+    To (reference temperature, typically 298.15 K), and R (universal gas constant).
+"""
+function compute_Kmm(
+    T::FT, 
+    p::FT, 
+    Kc25::FT, 
+    Ko25::FT, 
+    ΔHkc::FT, 
+    ΔHko::FT, 
+    To::FT, 
+    R::FT
+) where {FT}
+    Kc = MM_Kc(Kc25, ΔHkc, T, To, R)
+    Ko = MM_Ko(Ko25, ΔHko, T, To, R)
+
+    return Kc * (FT(1.0) + po2(p) / Ko) 
```

Comment:
we do this very inconsistently in our code, but here you can do 1+po2(p)/Ko, and 1 will take the right float type (i.e., no need for FT()